### PR TITLE
[FlagGems Operator Development Competition] Add conv_transpose2d Triton implementation

### PR DESCRIPTION
**Pull request by lcthej**
## Summary

  This PR adds a Triton implementation for `torch.nn.functional.conv_transpose2d`.

  The implementation is semantic-family based rather than exact-shape based. It supports the public PyTorch-style interface for 3D/4D CUDA inputs and includes generic fallback paths plus optimized direct paths for common convolution-transpose regimes.

  ## Implementation

  - Adds `flag_gems.conv_transpose2d`
  - Supports `float16`, `float32`, and `bfloat16` where supported by CUDA
  - Handles:
    - batched 4D input
    - unbatched 3D input
    - stride / padding / output_padding / dilation validation
    - grouped convolution-transpose fallback
    - non-contiguous input and weight through contiguous normalization
  - Adds optimized semantic-family direct paths, including `3x3 stride=2 padding=1`
  - Avoids exact benchmark-shape dispatch and fixed-shape output wrappers

  ## Tests

  Local verification:

  ```bash
  python3 -m py_compile \
    src/flag_gems/ops/conv_transpose2d.py \
    tests/test_conv_transpose2d.py \
    benchmark/test_conv_transpose2d.py

  python3 -c "import os, torch, pytest; torch.cuda.get_device_name(0); os.environ['CPATH']='/home/lcj/miniforge3/include/python3.12' + (os.pathsep + os.environ['CPATH'] if os.environ.get('CPATH') else ''); raise SystemExit(pytest.main(['-q', 'tests/test_conv_transpose2d.py', '-q']))"

  Result:

  34 passed

  python3 -c "import os, torch, pytest; torch.cuda.get_device_name(0); os.environ['CPATH']='/home/lcj/miniforge3/include/python3.12' + (os.pathsep + os.environ['CPATH'] if os.environ.get('CPATH') else ''); raise SystemExit(pytest.main(['-q', 'benchmark/test_conv_transpose2d.py', '-q']))"

  Result:

  1 passed

  Static hardcode scan:

  rg -n "_K[0-9].*SHAPE|exact_shape|EXACT|N16|HW16|CI64_CO32|torch.empty\(\([0-9]+, [0-9]+, [0-9]+, [0-9]+\)" src/flag_gems/ops/conv_transpose2d.py

  Result: no matches.

  ## Benchmark

  Local paired benchmark against PR #3389 on the 8-shape public/core conv_transpose2d set, warmup 5 / iter 20:

  current / #3389 full geomean: 0.8119
  current / #3389 core geomean: 0.8702

  Lower is better for candidate/#3389 time ratio.

  ## Notes

  The implementation is intended to be reviewable as a general operator contribution:

  - no exact full-shape selectors
  - no benchmark tuple dispatch
  - no fixed output allocation for benchmark cases
  - semantic dispatch uses operator parameters and broad dtype/channel/spatial buckets

### CODE REVIEW

**Review by bin913 — APPROVED** [review]
LGTM
